src/ValueObject/BaseValueObject.php

```diff
@@ -57,18 +59,13 @@ public function getValue()
     }
 
     /**
-     * @note should be abstract
-     *
      * @param TValue|mixed $value
      *
      * @return TValue
      *
      * @throws InvalidArgumentException
      */
-    protected static function filter($value)
-    {
-        throw self::invalidValue('expected');
-    }
+    abstract protected static function filter($value);
```

Comment:
Why does it name filter? I'd like to rename it to `adjustValue` or `validateAndAdjustValue` or something like this.
```suggestion
    abstract protected static function validateAndAdjustValue($value);
```

Reply:
Method `filter` refers to <https://www.php.net/manual/en/function.filter-var.php> - in ideal world the implementation should be like:

```php
protected static function filter(mixed $value): int
{
    $intCandidate = filter_var($value, FILTER_VALIDATE_INT);
    if ($intCandidate === false) {
        throw self::invalidValue('int');
    }
    return $intCandidate;
}
```

Reply:
OK, i don't wanna waste my time, so keep it as you wish.